src/fast_account_id.rs

```diff
@@ -0,0 +1,212 @@
+//! A fast alternative to `near_sdk::AccountId` that is faster to use, and has a
+//! smaller Borsh serialization footprint.
+
+use std::{ops::Deref, rc::Rc, str::FromStr};
+
+use near_sdk::borsh::{BorshDeserialize, BorshSerialize};
+
+/// An alternative to `near_sdk::AccountId` that is faster to use, and has a
+/// smaller Borsh serialization footprint.
+///
+/// Limitations:
+///  - Does not implement `serde` serialization traits.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct FastAccountId(Rc<str>);
```

Comment:
@dndll Figured `Rc` should be sufficient since contracts are always single-threaded.